packages/expo-face-detector/ios/EXFaceDetector/EXFaceDetectorManager.m

```diff
@@ -55,40 +70,50 @@ - (void)setIsEnabled:(BOOL)newFaceDetecting
 {
   // If the data output is already initialized, we toggle its connections instead of adding/removing the output from camera session.
   // It allows us to smoothly toggle face detection without interrupting preview and reconfiguring camera session.
-  
-  if ([self isDetectingFaces] != newFaceDetecting) {
-    _faceDetecting = newFaceDetecting;
-    __weak EXFaceDetectorManager *weakSelf = self;
+  if ([self isDetectingFaceEnabled] != newFaceDetecting) {
+    _faceDetectionEnabled = newFaceDetecting;
+    UM_WEAKIFY(self);
     [self _runBlockIfQueueIsPresent:^{
-      __strong EXFaceDetectorManager *strongSelf = weakSelf;
-      if (strongSelf) {
-        if ([strongSelf isDetectingFaces]) {
-          if (strongSelf.dataOutput) {
-            [strongSelf _setConnectionsEnabled:true];
-          } else {
-            [strongSelf tryEnablingFaceDetection];
-          }
-        } else {
-          [strongSelf _setConnectionsEnabled:false];
-        }
+      UM_ENSURE_STRONGIFY(self);
+      if ([self isDetectingFaceEnabled] && ![self isFaceDetecionRunning]) {
+        [self tryEnablingFaceDetection];
       }
     }];
   }
 }
 
 - (void)updateSettings:(NSDictionary *)settings
 {
-  [self _updateOptionSettingForKey:GMVDetectorFaceMode withJSONValue:settings[modeKeyPath]];
-  [self _updateOptionSettingForKey:GMVDetectorFaceLandmarkType withJSONValue:settings[detectLandmarksKeyPath]];
-  [self _updateOptionSettingForKey:GMVDetectorFaceClassificationType withJSONValue:settings[runClassificationsKeyPath]];
+  FIRVisionFaceDetectorOptions* newOptions = [EXFaceDetectorUtils newOptions:self.faceDetectorOptions withValues:settings];
+  if(![EXFaceDetectorUtils areOptionsEqual:newOptions to:self.faceDetectorOptions])
+  {
+    self.faceDetectorOptions = newOptions;
+    [self _resetFaceDetector];
+  }
+  if([settings objectForKey:kMinDetectionIntervalMillis])
+  {
+    self.timeIntervalMillis = [settings[kMinDetectionIntervalMillis] longValue];
+  }
```

Comment:
Let's follow these two `{`,`}` rules:
1. you place `{` after line break after method signature,
2. you place `{` in the same line in any other language-level instruction
shall we? 😄